fabric-key-binding-api-v1/src/main/java/net/fabricmc/fabric/api/client/keybinding/v1/FabricKeyBinding.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.keybinding.v1;
+
+import java.util.Objects;
+import java.util.function.BooleanSupplier;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.client.keybinding.StickyFabricKeyBinding;
+import net.fabricmc.fabric.mixin.client.keybinding.KeyCodeAccessor;
+
+/**
+ * Expanded version of {@link KeyBinding} for use by Fabric mods.
+ *
+ * <p>*ALL* instantiated FabricKeyBindings should be registered in
+ * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
+ * </p>
+ *
+ * <pre><code>
+ * FabricKeyBinding.Builder builder = FabricKeyBinding.builder();
+ * FabricKeyBinding left = builder
+ * 			.id(new Identifier("example", "left"))
+ * 			.code(Keys.Left)
+ * 			.build();
+ * FabricKeyBinding right = builder
+ * 			.id(new Identifier("example", "right"))
+ * 			.code(Keys.Right)
+ * 			.build();
+ * KeyBindingRegistry.register(left);
+ * KeyBindingRegistry.register(right);
+ * </code></pre>
+ */
+public class FabricKeyBinding extends KeyBinding {
```

Comment:
final?

Reply:
need to extend this for v0.

Reply:
Any reasonable way to avoid this? Allowing sub classes may make future changes more difficult.

Reply:
Would making an interface for detecting if it is a fabric keybind be alright? need to do that for it to work rn, might look at how it works to tweak it

Reply:
If there's no simple solution we can leave it not final

Reply:
Made an interface called `ModdedKeyBinding` for both the v1 and v0 keybinding to extend, I can't think of a better name for that class.

Reply:
nvm I reverted this.

Reply:
We might be above to just never return the fabric version of the keybind to the player, and return the vanilla ones, I don't see a clear reason how getIdentitier() is useful, vanilla keybindings don't have them anyways